wcfsetup/install/files/lib/system/image/adapter/GDImageAdapter.class.php

```diff
@@ -493,6 +493,47 @@ public function overlayImageRelative($file, $position, $margin, $opacity) {
 		// does nothing
 	}
 	
+	/**
+	 * @inheritDoc
+	 */
+	public function saveImageAs($image, string $filename, string $type, int $quality = 100): void {
+		if (!$this->isImage($image)) {
+			throw new SystemException("Given image is not a valid image resource.");
```

Comment:
Do not use `SystemException` for entirely new code that does not need the backwards compatibility with catch blocks.

 `InvalidArgumentException` would be fitting here.